src/lib/sitemap/tutorials-content-fields.ts

```diff
@@ -0,0 +1,50 @@
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import { SectionOption } from 'lib/learn-client/types'
+import { activeProductSlugs } from 'lib/products'
+import { generateTutorialMap } from 'pages/api/tutorials-map'
+import { ProductSlug } from 'types/products'
+import { getCollectionSlug } from 'views/collection-view/helpers'
+import { makeSitemapField } from './helpers'
+import { Collection as ClientCollection } from 'lib/learn-client/types'
+
+function getTutorialLandingPaths(): string[] {
+	const activeSlugs = activeProductSlugs.map(
+		(productSlug: ProductSlug) => `/${productSlug}/tutorials`
+	)
+	const sectionOptionsWithLandingPage = Object.values(SectionOption).filter(
+		(option: SectionOption) => option === SectionOption.onboarding
+	)
+	const sectionOptions = sectionOptionsWithLandingPage.map(
+		(slug: SectionOption) => `/${slug}`
+	)
+
+	return [...activeSlugs, ...sectionOptions]
+}
+
+async function getCollectionPaths() {
+	const allCollections = await getAllCollections()
+
+	return allCollections.map((collection: ClientCollection) => {
+		// build collection paths
+		return getCollectionSlug(collection.slug)
+	})
+}
+
+export async function allTutorialsFields() {
+	const landingSlugs = getTutorialLandingPaths()
+	const tutorialLandingFields = landingSlugs.map((slug: string) =>
+		makeSitemapField({ slug })
+	)
+
+	const collectionSlugs = await getCollectionPaths()
+	const collectionFields = collectionSlugs.map((slug: string) =>
+		makeSitemapField({ slug })
+	)
+
+	const tutorialSlugs = Object.values(await generateTutorialMap())
+	const tutorialFields = tutorialSlugs.map((slug: string) =>
+		makeSitemapField({ slug })
+	)
+
+	return [...tutorialFields, ...collectionFields, ...tutorialLandingFields]
```

Comment:
```suggestion
	const landingSlugs = getTutorialLandingPaths()
	const collectionSlugs = await getCollectionPaths()
	const tutorialSlugs = Object.values(await generateTutorialMap())

	return [...landingSlugs, ...collectionSlugs, ...tutorialSlugs].map(
		(slug: string) => makeSitemapField({ slug })
	)
```
💡 take it or leave it! Just saw an opportunity to dry it up a bit.

Reply:
Oh I like it!